libraries/helpers.rb

```diff
@@ -6,30 +6,36 @@ def type_from_name
     end
 
     # Pluralize object directory name
+    # gsub because English is weird
     def object_dir
-      ::File.join(new_resource.config_home, type_from_name) + 's'
+      ::File.join(new_resource.config_home, type_from_name).gsub('y', 'ie') + 's'
```

Comment:
We need to bound the regex to only replace the last character of `y` otherwise if there is another `y` in the path this will not behave as expected. I think this would work:
```
#{::File.join(new_resource.config_home, type_from_name).gsub(/y$/, 'ie')}s
```
A couple of ruby nitpicks:
- using string concatenation creates `n+1` strings (in this case 3) where string interpolation only uses a single string and is just as fast with less resources required
- if your not using regex I think `tr` would be better/faster

Granted in this use case neither of the above optimizations make any real difference but I have seen it make a big difference in large long running ruby applications.

Reply:
Sounds reasonable, this could definitely be cleaner. I could also just revert for now and not include this change. Seeing an `entitys` folder just felt wrong but functionally makes no difference. I can make it in an isolated PR if preferred.

Reply:
I think I'll refactor this out, kind of a silly change to bury in this PR given the size of the rest of changes.

Reply:
@majormoses reverted to previous functionality in latest commit. I agree with the potential string concat `n+1` issue and will address that in a different, smaller PR.